src/python/pants/util/resources.py

```diff
@@ -0,0 +1,37 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+
+import importlib
+from itertools import chain
+
+import importlib_resources as resources
+
+
+def read_resource(package_or_module: str, resource: str) -> bytes:
+    """Reads a resource file from within the Pants package itself.
+
+    This helper function is designed for compatibility with `pkgutil.get_data()` wherever possible,
+    but also allows compability with PEP302 pluggable importers such as included with PyOxidizer.
+    This requires that resources are loaded from a valid Python package (i.e. must have an
+    `__init__.py` file in the directory).
+    """
+
+    a = importlib.import_module(package_or_module)
+    package_ = a.__package__
+
+    if package_ is None:
+        raise ValueError(
+            "`read_resource` can only help find resources for packages or modules that live in "
+            "a package."
+        )
+
+    resource_parts = resource.split("/")
```

Comment:
Maybe make resource `str | PurePath` and convert to `PurePath` if not and leverage `.parts` - Windows and all.

Reply:
I think using `/` is correct. If I'm reading [this](https://docs.python.org/3/library/pkgutil.html#pkgutil.get_data) correctly, pkg_util, whose behavior we're emulating here, resource paths always use a forward slash on all platforms.

AFAICT in importlib resource names must not contain a path separator at all, but be directly in the specified package, hence the dance here.

Reply:
Aha, you're right.